core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/DataSourceWizardService.java

```diff
@@ -166,6 +166,9 @@ public void removeDSW( String dswId ) throws PentahoAccessControlException {
   }
 
   public List<String> getDSWDatasourceIds() {
+    if ( dataSourceAwareMetadataDomainRepository != null ) {
```

Comment:
allowing for backwards compatibility, if metadataRepository doesn't support the new interface introduced in (https://github.com/pentaho/pentaho-platform/blob/b5939dae950f412abf0b6d18dcd01ec1debb9e5e/extensions/src/main/java/org/pentaho/platform/plugin/services/metadata/IDataSourceAwareMetadataDomainRepository.java) then fall back to previous implementation.